### ISSUE TITLE
2.4.0 Ambient Context Accessor Build Failures

### ISSUE THREAD

**Issue by vorion** [peterfreeman/ngx-dropzone#94]
Edit: Seems like 2.4.0 "minor" update is for ng11 support. Many installs using ^2.X will need to update to ~2 if on older ng.

================================

Builds are failing for 2.4.0 due to ambient acessors

` ng build --aot=false --buildOptimizer=false --optimization=false --configuration=dev`

ERROR in node_modules/ngx-dropzone/lib/ngx-dropzone-preview/ngx-dropzone-preview.component.d.ts:9:9 - error TS1086: An accessor cannot be declared in an ambient context.

9     get removable(): boolean;
          ~~~~~~~~~
node_modules/ngx-dropzone/lib/ngx-dropzone/ngx-dropzone.component.d.ts:30:9 - error TS1086: An accessor cannot be declared in an ambient context.

30     get maxFileSize(): number;
           ~~~~~~~~~~~
node_modules/ngx-dropzone/lib/ngx-dropzone/ngx-dropzone.component.d.ts:31:9 - error TS1086: An accessor cannot be declared in an ambient context.

31     set maxFileSize(value: number);
           ~~~~~~~~~~~
node_modules/ngx-dropzone/lib/ngx-dropzone/ngx-dropzone.component.d.ts:34:9 - error TS1086: An accessor cannot be declared in an ambient context.

34     get expandable(): boolean;
           ~~~~~~~~~~
> XXXXXXXXX@1.0.0 build-dev D:\a\1\s
node_modules/ngx-dropzone/lib/ngx-dropzone/ngx-dropzone.component.d.ts:35:9 - error TS1086: An accessor cannot be declared in an ambient context.
> ng build --aot=false --buildOptimizer=false --optimization=false --configuration=dev

**Comment by hikaru150s** [comment]
For anyone (including any package creator) who create packages for angular, please take cautious with **Angular version** and/or **Typescript Version**. My company was heavily dependent to this package and it was shocking when we doing annual package maintenance and found that current version of `ngx-dropzone` (2.4.0) was built on top of Angular 9+.

To be noted, there was a compiler difference between Angular 2+~8 with 9+ (Ivy), and Angular 11 implied to use Typescript 4 which make it even worse. As a result, when user try to run a compiled-version of TS4 into TS3 environment will cause several issues (like this one).

As for this issue, I can't help much except to downgrade your version. I think version 2.3.0 is the last I known to be compatible for Angular 2+~8. Need more research for this.

And for package maintainer, please give us some notes about compatibilities (at least, please give us notes that version 2.4.0 is only compatible with Angular 11+ or maybe you can give semver `^2.0.0` for Angular 2+~8 and `^3.0.0` for Angular 11+)
Thank you very much.

**Comment by vorion** [comment]
@hikaru150s Indeed. We were able to fallback on ~2.1.2 without issue for Angular8 instead of ^2.1.2 which forced us up to 2.4.0

**Comment by peterfreeman** [comment]
Hey guys,

I reverted the upgrade to Angular 11 and released v2.5.0, so all of your CI workflows should run again.
I then re-applied the upgrade and released v3.0.0 which is now using Angular 11.

This should fix any incompatibilities, sorry for the inconveniences.

Happy coding! 🤓
